### ISSUE TITLE
pd-merge cross product doesn't produce a cross product

### ISSUE THREAD

**Issue by genmeblog** [techascent/tech.ml.dataset#289]
```
(require '[tech.v3.dataset.join :as j])
(def a (tc/dataset {:a [1 2 3] :b [4 5 6]}))
(def b (tc/dataset {:c [:a :b :c] :d [:x :y :z]}))

(j/pd-merge a b {:how :cross})
;; => cross-join [9 4]:
;;    | :a | :b | :c | :d |
;;    |---:|---:|----|----|
;;    |  1 |  4 | :a | :x |
;;    |  2 |  5 | :b | :y |
;;    |  3 |  6 | :c | :z |
;;    |  1 |  4 | :a | :x |
;;    |  2 |  5 | :b | :y |
;;    |  3 |  6 | :c | :z |
;;    |  1 |  4 | :a | :x |
;;    |  2 |  5 | :b | :y |
;;    |  3 |  6 | :c | :z |
```

**Comment by cnuernber** [comment]
Yes that is wrong.  There should not be duplicate rows.

**Comment by cnuernber** [comment]
Fixed in latest release (6.066) - 

```clojure
tech.v3.dataset.join-test> (ds-join/pd-merge
             (ds/->dataset {:a [1 2 3] :b [4 5 6]})
             (ds/->dataset {:c [:a :b :c] :d [:x :y :z]})
             {:how :cross})
cross-join [9 4]:

| :a | :b | :c | :d |
|---:|---:|----|----|
|  1 |  4 | :a | :x |
|  1 |  4 | :b | :y |
|  1 |  4 | :c | :z |
|  2 |  5 | :a | :x |
|  2 |  5 | :b | :y |
|  2 |  5 | :c | :z |
|  3 |  6 | :a | :x |
|  3 |  6 | :b | :y |
|  3 |  6 | :c | :z |
```
